src/main/java/org/truffleruby/builtins/CoreMethodNodeManager.java

```diff
@@ -150,7 +150,7 @@ public void addLazyCoreMethod(String nodeFactoryName, String moduleName, Visibil
 
         final SharedMethodInfo sharedMethodInfo = makeSharedMethodInfo(context, module, required, optional, rest, names[0]);
 
-        final RubyNode methodNode = new LazyRubyNode(() -> {
+        final RubyNode methodNode = new LazyRubyNode(() -> sharedMethodInfo.getSourceSection(), () -> {
```

Comment:
Isn't it enough that the RubyRootNode has a SourceSection?
They even have the same SourceSection here.

Reply:
No it isn't.